File: Problemas/1368A - C+=/1368A_C+=.cpp
// 1368A - C+=
// https://codeforces.com/problemset/problem/1368/A

#include <stdio.h>
#include <stdlib.h>
#include <bits/stdc++.h>
#include <cmath>
using namespace std;

int main(){

    // Input:
    // 1) pruebas = Número de casos de prueba
    int pruebas;
    cin >> pruebas;

    // Itera sobre cada caso de prueba
    for(int indice_prueba = 0; indice_prueba < pruebas; indice_prueba++) {

        // Input:
        // 2) a,b = Números iniciales
        // 3) n = Número a superar
        int a,b,n;
        cin >> a >> b >> n;

        // Settea los primeros dos términos del Fibonacci
        int t1, t2;
        if(a <= b) {
            t1 = a;
            t2 = b;
        } else {
            t1 = b;
            t2 = a;
        }

        // Variable para autosuma
        int suma = 0;

        // Contador de números de Fibonacci
        int contador = 0;

        // Mientras la autosuma no exceda N, expande el Fibonacci
        while(suma <= n){

            // Realiza la autosuma
            suma = t1 + t2;
            t1 = t2;
            t2 = suma;

            // Lleva el conteo de números de Fibonacci generados
            contador++;
        }
        cout << contador << endl;

    }

    return 0;
}

File: Problemas/1368A - C+=/1368A_C+=.py
# 1368A - C+= 
# https://codeforces.com/problemset/problem/1368/A

# Input:
# 1) pruebas = Número de casos de prueba
pruebas = int(input())

# Itera sobre cada caso de prueba
for indice_prueba in range(0,pruebas):

    # Input:
    # 2) a,b = Números iniciales
    # 3) n = Número a superar
    a,b,n = map(int, input().split())

    # Secuencia de Fibonacci
    fibonacci = [a, b]

    # Ordena a y b de menor a mayor
    fibonacci = sorted(fibonacci)  
    
    # Variable para autosuma
    suma = 0

    # Mientras la autosuma no exceda N, expande el Fibonacci
    while(suma <= n):

        # Realiza la autosuma
        suma = fibonacci[len(fibonacci)-1] + fibonacci[len(fibonacci)-2]

        # Añade la autosuma al Fibonacci
        fibonacci.append(suma)

    # Imprime la cantidad de números generada en la secuencia de Fibonacci
    print(len(fibonacci) - 2)